chia/data_layer/data_store.py

```diff
@@ -659,6 +669,32 @@ async def add_kvid(self, blob: bytes, store_id: bytes32, writer: aiosqlite.Conne
             path.write_bytes(zstd.compress(blob))
         return KeyOrValueId(row[0])
 
+    def delete_unconfirmed_kvids(self, store_id: bytes32) -> None:
+        for blob_hash in self.unconfirmed_keys_values[store_id]:
+            with log_exceptions(log=log, consume=True):
+                path = self.get_key_value_path(store_id=store_id, blob_hash=blob_hash)
+                try:
+                    path.unlink()
+                except FileNotFoundError:
+                    log.error(f"Cannot find key/value path {path} for hash {blob_hash}")
+        self.unconfirmed_keys_values[store_id].clear()
+
+    def confirm_all_kvids(self, store_id: bytes32) -> None:
+        self.unconfirmed_keys_values[store_id].clear()
+
+    @contextmanager
+    def manage_kv_files(self, store_id: bytes32) -> Iterator[None]:
+        if self.unconfirmed_keys_values[store_id]:
```

Comment:
```suggestion
        d = {}
        if self.unconfirmed_keys_values.setdefault(store_id, d) is not d:
```

or, a bit less fancy pants...

```suggestion
        if store_id not in self.unconfirmed_keys_values:
            self.unconfirmed_keys_values[store_id] = {}
        else:
```